modules/top-level/test.nix

```diff
@@ -1,10 +1,90 @@
-{ lib, ... }:
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+let
+  cfg = config.test;
+in
 {
   options.test = {
+    name = lib.mkOption {
+      type = lib.types.str;
+      default = "nixvim-check";
+      description = "The test derivation's name.";
+    };
+
     runNvim = lib.mkOption {
       type = lib.types.bool;
       description = "Whether to run `nvim` in the test.";
       default = true;
     };
+
+    # Output
+    derivation = lib.mkOption {
+      type = lib.types.package;
+      description = ''
+        A derivation that tests the config by running neovim.
+      '';
+      readOnly = true;
+    };
+  };
+
+  config = {
+    test.derivation = pkgs.stdenv.mkDerivation {
+      inherit (cfg) name;
+      dontUnpack = true;
+
+      nativeBuildInputs = [
+        config.finalPackage
+        # TODO: Do we really need docker?
```

Comment:
Do we know this now?

Reply:
It seems [to build fine](https://buildbot.nix-community.org/#/builders/26/builds/436) without it